client/src/components/TagField.js

```diff
@@ -11,38 +11,60 @@ import url from 'url';
 import debounce from 'debounce-promise';
 import PropTypes from 'prop-types';
 
-class TagField extends Component {
-  constructor(props) {
-    super(props);
+const TagField = ({
+  lazyLoad = false,
+  options,
+  creatable = false,
+  multi = false,
+  disabled = false,
+  labelKey = 'Title',
+  valueKey = 'Value',
+  SelectComponent = Select,
+  AsyncCreatableSelectComponent = AsyncCreatableSelect,
+  AsyncSelectComponent = AsyncSelect,
+  CreatableSelectComponent = CreatableSelect,
+  value: initialValue,
```

Comment:
What's this colon syntax doing here, compared with `=` syntax? And where does this `initialValue` come from? This wasn't in default props.

Reply:
tl;dr Fancy syntax so that we can use the nicely name `value` as something we can reassign inside the component would otherwise clash with the prop

It's basically shorthand for

```
function TagField(props) {
  const initialValue = props.value; // immutable
  let value = props.value; // mutable - gets reassigned in the functional component
```

We're using the modern `const TagField({...}) {` syntax though so that we can use Javascript default parameters, so we're using object deconstruction and aliasing to achieve the same thing i.e. not using the `props` variable

Reply:
Thanks for explaining, that makes sense.